chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_bubble_view.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_BRAVE_TAB_HOVER_CARD_BUBBLE_VIEW_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_BRAVE_TAB_HOVER_CARD_BUBBLE_VIEW_H_
+
+#define BRAVE_TAB_HOVER_CARD_BUBBLE_VIEW_H_ \
+                                            \
+ protected:                                 \
+  class BraveAnonymous;                     \
```

Comment:
I'm not understanding what is happening here. Why are we creating an inner class that just has a single static method? Why do we need a class here at all?

Reply:
I am using an inner class here because inner classes have access to private members.
Because I am redefining `TabHoverCardBubbleView` as `TabHoverCardBubbleView_ChromiumImpl` (and then declaring the subclass as `TabHoverCardBubbleView`) I can't add a `friend class TabHoverCardBubbleView`.